### PR TITLE
Improve test coverage for i_node.cc

### PR DESCRIPTION
**Pull request by st-pasha**
- Many tests in `test_dt_rows.py` used old-fashioned style `DT(rows=...)` of selecting elements from a Frame. Consequently, those tests were not running for the new-style selectors `DT[i, j]` expressions. As a result, several bugs were inadvertently introduced during the latest round of refactoring, as the tests effectively ignored the new code. This PR addresses this issue, and transforms all tests in `test_dt_rows.py` into the "new-style" selectors of the form `DT[i, j]`. Eventually the old-style calling method `DT(...)` will be deprecated and later removed.

- After the tests were enabled, few more bugs were discovered and fixed, in particular bugs involving "multislice" `i` selectors;

- Fixed coverage script `llvm-gcov.sh`;

- Handle correctly all reducer nodes, including sd, min, max and mean;

- Several tests were added in order to achieve 100% coverage of `i_node.cc`;

- In some tests error messages were improved.

### CODE REVIEW

File: c/rowindex_array.cc [hunk 1]
```diff
@@ -403,7 +403,7 @@ void ArrayRowIndexImpl::init_from_integer_column(const Column* col) {
   }
   int64_t imin = col->min_int64();
   int64_t imax = col->max_int64();
-  if (imin < 0) {
+  if (imin < -1) {
```
Comment (oleksiyskononenko) [inline c/rowindex_array.cc]: So `imin` is allowed to be `-1`?

File: ci/llvm-gcov.sh [hunk 2]
```diff
@@ -1,2 +1,21 @@
 #!/bin/bash
-exec "$LLVM4$LLVM5$LLVM6/bin/llvm-cov" gcov "$@"
+
+if [ -n "$LLVM7" ]; then
```
Comment (oleksiyskononenko) [inline ci/llvm-gcov.sh]: Should we update documentation online saying that LLVM7 is also supported? I guess I checked it yesterday, and it was only 4/5/6.

File: c/rowindex_array.cc (hunk 1 above)
Comment (oleksiyskononenko) [reply c/rowindex_array.cc]: Ah, this seems to be connected to #1496?

Comment (st-pasha) [reply c/rowindex_array.cc]: yes

File: ci/llvm-gcov.sh (hunk 2 above)
Comment (st-pasha) [reply ci/llvm-gcov.sh]: Well, we never tested with LLVM7, so it's too early to claim support. However, eventually when we add the support, this script won't have to be modified.

**Pull request merged**